src/api/context/context.zig

```diff
@@ -0,0 +1,430 @@
+///
+/// ## Returns
+/// `true` if the context was successfully detached.
+pub fn detachContext(token: Token) DetachError!bool {
+    const stack = context_stack orelse return error.ContextStackNotInitialized;
+    return stack.detach(token);
+}
+
+/// Cleans up all thread-local context resources.
+///
+/// This function should be called before thread termination to prevent
+/// memory leaks. It releases the context stack and all attached contexts
+/// for the current thread.
+///
+/// ## Thread Safety
+/// This function only affects the calling thread's context stack.
+///
+/// ## Usage
+/// ```zig
+/// defer context.cleanup(); // At thread/program exit
+/// ```
+pub fn cleanup() void {
+    if (context_stack) |stack| {
+        const allocator = stack.contexts.allocator;
+        stack.deinit();
+        allocator.destroy(stack);
+        context_stack = null;
+    }
+}
+
+// --- Tests ---
+test "key creation uniqueness" {
+    const key1 = createKey("test");
+    const key2 = createKey("test");
+    try std.testing.expect(key1.id != key2.id);
+}
+
+test "context immutability" {
+    const allocator = std.testing.allocator;
+    const ctx1 = Context.init();
+    const key = createKey("test_key");
+    var ctx2 = try ctx1.setValue(allocator, key, .{ .string = "value1" });
+    defer ctx2.deinit();
+    try std.testing.expect(ctx1.getValue(key) == null);
+    const value = ctx2.getValue(key).?;
+    try std.testing.expectEqualStrings("value1", value.string);
+}
+
+test "context operations" {
+    const allocator = std.testing.allocator;
+    defer cleanup();
+
+    try std.testing.expect(getCurrentContext().entries == null);
+
+    const key = createKey("test");
+    const ctx = try Context.init().setValue(allocator, key, .{ .int = 42 });
+    const token = try attachContext(ctx);
+    const value = getCurrentContext().getValue(key).?;
+
+    try std.testing.expectEqual(@as(i64, 42), value.int);
+    try std.testing.expect(try detachContext(token));
+    try std.testing.expect(getCurrentContext().getValue(key) == null);
+}
+
+test "detach error handling" {
+    const allocator = std.testing.allocator;
+    defer cleanup();
+
+    const ctx1 = try Context.init().setValue(allocator, createKey("k1"), .{ .int = 1 });
+    const ctx2 = try Context.init().setValue(allocator, createKey("k2"), .{ .int = 2 });
+
+    const token1 = try attachContext(ctx1);
+    const token2 = try attachContext(ctx2);
+
+    try std.testing.expectError(error.DetachOrderError, detachContext(token1));
+
+    try std.testing.expect(try detachContext(token2));
+    try std.testing.expect(try detachContext(token1));
+}
+
+test "compile time key creation" {
+    const MyKey = Key("my_service.request_id");
+    const OtherKey = Key("my_service.request_id");
+    try std.testing.expect(MyKey.id != OtherKey.id);
+    try std.testing.expectEqual(@as(usize, 0), MyKey.id);
+    try std.testing.expectEqual(@as(usize, 1), OtherKey.id);
+}
```

Comment:
Why is the test named "compile time"?
Don't we need to explicitly mark as `comptime` the assignments and function call to make them happen at comptime?